test/gost2814789test.c
=======================================================
lhs: 100644 | b2cd41fadc183dc5bc8fe680c63f4bf37ca1b6e9
rhs: 100644 | 953e1e154046d6b9df8960ab7b474b9d80eb83a7
---@@ -1411,6 +1411,7 @@ int main(int argc, char *argv[])
             }
             siglen = 4;
             OPENSSL_assert(EVP_DigestSignFinal(&mctx, bTest, &siglen));
+            EVP_PKEY_free(mac_key);
             EVP_MD_CTX_cleanup(&mctx);
             enlu = (int)tcs[t].ullLen;
             enlf = 0;
@@ -1434,6 +1435,8 @@ int main(int argc, char *argv[])
     printf(" passed\n");
     fflush(NULL);
 
+    NCONF_free(pConfig);
+
     return EXIT_SUCCESS;
 }
 #endif

---